src/odemis/gui/util/img.py

```diff
@@ -1888,7 +1888,9 @@ def get_ordered_images(streams, raw=False):
             if numpy.can_cast(im_min_type, min_type(data_raw)):
                 im_min_type = min_type(data_raw)
 
-            if data_raw.ndim > 2:
+            if isinstance(s, DataProjection):
```

Comment:
I don't think this is related to the line-spectrum export, as this is only called from `images_to_export_data()`, which is for the "spatial" views.
Anyway, I think you're on the right way. Indeed, you'd need to handle when the stream `s` is a projection. Actually, I don't think it should even be needed to handle the case of receiving a `SpectrumStream`. There are probably "only" three cases:
 * `s` is a static stream, which always has a .raw, in which case the "standard" lines from 1879 should be applied
 * `s` is a projection, and the s.stream is *not* a SpectrumStream, in which case `.raw` should be used again, and the "standard" lines should be applied
  * `s` is a projection and the s.stream is a `SpectrumStream`, in which case `.projectAsRaw()` should be used.

I think that ideally whenever it's a projection, it should be possible to use `.projectAsRaw()`, or that `.projectAsRaw()` should be only available on the projections where it does properly return the raw data.  Both could be quite easy to implement (as just RGBSpatialProjection doesn't provide a proper `.projectAsRaw()`).  For now, the logic described above could work, but it might be easier to just implement `.projectAsRaw()` entirely, and keep the code simple here.